src/highlight.js

```diff
@@ -819,8 +819,18 @@ const HLJS = function(hljs) {
    * auto-highlights all pre>code elements on the page
    */
   function highlightAll() {
+    function boot() {
+      // if a highlight was requested before DOM was loaded, do now
+      highlightAll();
+      window.removeEventListener('DOMContentLoaded', boot, false);
```

Comment:
What is the thinking here?  Is this type of cleanup truly necessary?

Reply:
This cleanup might only matter in rare cases, like when an `hljs` instance is used just once and `highlightAll` is called before `DOMContentLoaded`. In such a scenario, the instance would stay in memory, unable to be released. If this corner case isn't a concern, the cleanup is likely unnecessary.

If this contradicts the core philosophy, the cleanup can be removed.

Reply:
I'm just going to call it unnecessary -- unless you have some docs (from MDN or such) that says this is somehow best practice... I've seen tons of `onload` hooks like this over the years but never code to unwind a single-use event listner afterwards - seems a browser concern... if the browser knows that event that just fired can never be fired again on the same page it could do the cleanup itself.

Reply:
I understand your point. While the event could technically be triggered using `window.dispatchEvent(new Event("DOMContentLoaded"))`, which isn't common practice, this means the browser wouldn’t automatically handle the cleanup. However, another approach to ensure the listener is only triggered once and automatically cleaned up is to use `window.addEventListener(..., { once: true })`. Is this one better, or just remove the cleanup?

Would you prefer switching to `{ once: true }`, or should we remove the cleanup entirely?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event

I find zero examples of `once` being suggested.  Lets just remove the cleanup.

Reply:
Thanks for the clarification and patience! I've removed the cleanup.